src/questionnaireitem_system_focus.js

```diff
@@ -0,0 +1,58 @@
+/**
+A QuestionnaireItemSystem that stores how often and how long the focus was moved away from the page.
+
+Reports as array in milliseconds how long the focus was lost each time.
+
+@class QuestionnaireItemSystemFocus
+@augments UIElement
+@augments UIElementInteractive
+@augments QuestionnaireItem
+@augments QuestionnaireItemSystem
+*/
+function QuestionnaireItemSystemFocus() {
+    QuestionnaireItemSystem.call(this, null, "Focus", false);
+
+    this.focusLossDurations = [];
+    this.answer = this.focusLossDurations;
+    this.timeOfLastFocusLoss = new Date().getTime();
+    this.inFocus = null;
+}
+QuestionnaireItemSystemFocus.prototype = Object.create(QuestionnaireItemSystem.prototype);
+QuestionnaireItemSystemFocus.prototype.constructor = QuestionnaireItemSystemFocus;
+
+QuestionnaireItemSystemFocus.prototype.createUI = function() {
+    this.onlostfocus = (this._onlostfocus).bind(this);
+    window.addEventListener("blur", this.onlostfocus, false);
+
+    this.ongainedfocus = (this._ongainedfocus).bind(this);
+    window.addEventListener("focus", this.ongainedfocus, false);
+};
+
+QuestionnaireItemSystemFocus.prototype.releaseUI = function() {
+    window.removeEventListener("blur", this.onlostfocus, false);
+    window.removeEventListener("focus", this.ongainedfocus, false);
+    this.answer = this.focusLossDurations;
+};
+QuestionnaireItemSystemFocus.prototype._onlostfocus = function() {
+    // Blur event can be triggered multiple times in a row
+    if (this.inFocus == null || this.inFocus == true) {
+        this.inFocus = false;
+        this.timeOfLastFocusLoss = new Date().getTime();
+    }
+};
+QuestionnaireItemSystemFocus.prototype._ongainedfocus = function() {
+    if (this.inFocus == null || this.inFocus == false) {
+        this.inFocus = true;
+        this.focusLossDurations.push(new Date().getTime() - this.timeOfLastFocusLoss);
+        this.answer = this.focusLossDurations;
+    }
+};
+QuestionnaireItemSystemFocus.prototype.getData = function() {
+    return [this.getQuestion(), this.getAnswer()];
+};
+QuestionnaireItemSystemFocus.prototype.isReady = function() {
+    return true;
+};
+QuestionnaireItemSystemFocus.prototype.getData = function() {
+    return this.focusLossDurations;
```

Comment:
Implementation of setData() is missing.
It is actually not used at the moment, but will be when localstorage is re-implemented.